Mathlib/NumberTheory/DirichletCharacter/Basic.lean

```diff
@@ -41,39 +40,53 @@ Main definitions:
 dirichlet character, multiplicative character
 -/
 
+/-!
+### Definitions
+-/
+
 /-- The type of Dirichlet characters of level `n`. -/
 abbrev DirichletCharacter (R : Type*) [CommMonoidWithZero R] (n : ℕ) := MulChar (ZMod n) R
 
 open MulChar
+
 variable {R : Type*} [CommMonoidWithZero R] {n : ℕ} (χ : DirichletCharacter R n)
 
 namespace DirichletCharacter
+
 lemma toUnitHom_eq_char' {a : ZMod n} (ha : IsUnit a) : χ a = χ.toUnitHom ha.unit := by simp
 
 lemma toUnitHom_eq_iff (ψ : DirichletCharacter R n) : toUnitHom χ = toUnitHom ψ ↔ χ = ψ := by simp
 
 lemma eval_modulus_sub (x : ZMod n) : χ (n - x) = χ (-x) := by simp
 
-lemma periodic {m : ℕ} (hm : n ∣ m) : Function.Periodic χ m := by
-  intro a
-  rw [← ZMod.natCast_zmod_eq_zero_iff_dvd] at hm
-  simp only [hm, add_zero]
```

Comment:
_Note to reviewers of this PR_: I removed this lemma, and the import needed to state it, because it doesn't actually do anything. The Dirichlet character `χ` coerces to a function `ZMod n → R` so `m` is getting coerced into `ZMod n`; and the hypotheses imply that `m` coerces to zero in `ZMod n`. So this is actually showing that `χ` is periodic with period 0, which is not a useful statement. 

(Maybe `Function.Periodic (χ ∘ Int.cast) m` is what was intended, but it is probably better to show that via periodicity of `Int.cast`.)